components/organisms/UserSettingsPage/user-settings-page.tsx

```diff
@@ -338,10 +338,11 @@ const UserSettingsPage = ({ user }: UserSettingsPageProps) => {
             </div>
 
             <div id="upgrade" className="flex flex-col gap-2">
-              <label className="flex flex-col w-full gap-2">
+              <label htmlFor="timezone-select" className="flex flex-col w-full gap-2">
```

Comment:
`<label htmlFor="..." />` is definitely a good practice, but in this particular case and the other places where you made similar changes, the label wraps the form input, so they are already connected, so `htmlFor` and `id` aren't necessary. It might be worth just bringing in the `<label />` into the component and have a `labelText` prop.

Reply:
please clear me here;

I should move the label element into the component e.g. in `components/atoms/Select/single-select.tsx` but there are other places where the label is used without having an input element or component like TextInput e.g. 

```typescript
<form
                name="delete-account"
                action="/api/delete-account"
                method="POST"
                className="flex flex-col order-first gap-6 md:order-last p-6 rounded-2xl bg-light-slate-4"
                ref={deleteFormRef}
                onSubmit={(e) => {
                  setIsModalOpen(true);
                  e.preventDefault();
                }}
              >
                <div className="flex flex-col gap-3">
                  <label htmlFor="delete-account-button" className="text-2xl font-normal">
                    Delete Account
                  </label>
                  <div className="w-full md:w-96">
                    <Text>
                      Please note that account deletion is irreversible. Proceed only if you are certain about this
                      action.
                    </Text>
                  </div>
                </div>
                <Button
                  id="delete-account-button"
                  type="submit"
                  rel="noopener noreferrer"
                  target="_blank"
                  variant="destructive"
                  className="w-max"
                >
                  Delete Account
                </Button>
                <DeleteAccountModal
                  open={isModalOpen}
                  setOpen={setIsModalOpen}
                  onDelete={() => {
                    setIsModalOpen(false);
                    deleteFormRef.current?.submit();
                  }}
                />
              </form>
  ```

Reply:
Tagging you in case you missed it @nickytonline

Reply:
For that particular example, see https://github.com/open-sauced/app/pull/4152/files#r1792034100

Reply:
@nickytonline I want to point out that

- eslint gives a warning when there is input or select( or any) custom component placed inside the label element, 

please see the changes made in select and single-select components along with one example where they have been used fine or not

 `components/atoms/Select/single-select.tsx`?
```typescript

interface SelectTriggerProps extends React.ComponentPropsWithoutRef<typeof SelectPrimitive.Trigger> {
  selectIcon?: React.ReactNode;
  labelText?: string;
}

const SelectTrigger = React.forwardRef<React.ElementRef<typeof SelectPrimitive.Trigger>, SelectTriggerProps>(
  ({ className, children, selectIcon, labelText, ...props }, ref) => (
    <label className="flex flex-col w-full text-sm">
      {labelText && <span className="sr-only">{labelText}</span>}
      <SelectPrimitive.Trigger
        ref={ref}
        className={clsx(
          "flex min-h-[2.1rem] items-center radix-select-trigger justify-between rounded-lg overflow-hidden  px-3 py-1.5 text-sm border  focus:outline-none  disabled:cursor-not-allowed disabled:opacity-50",
          className
        )}
        {...props}
      >
        {children}
        <SelectPrimitive.Icon asChild>{selectIcon}</SelectPrimitive.Icon>
      </SelectPrimitive.Trigger>
    </label>
  )
);

SelectTrigger.displayName = SelectPrimitive.Trigger.displayName;

```

changes made in `components/organisms/UserSettingsPage/user-settings-page.tsx`

```typescript
<div id="upgrade" className="flex flex-col gap-2">
              <div className="flex flex-col w-full gap-2">
                Time zone
                <Select onValueChange={(value) => setTimezone(value)} value={timezone}>
                  <SelectTrigger
                    labelText="Time zone"
                    selectIcon={
                      <div className="relative pr-4">
                        <RiArrowUpSLine size={16} className="absolute -top-3" />
                        <RiArrowDownSLine size={16} className="absolute -bottom-3" />
                      </div>
                    }
                  >
                    <SelectValue placeholder="Select time zone" />
                  </SelectTrigger>

                  <SelectContent position="item-aligned" className="bg-white">
                    <SelectItem value={""}>Select timezone</SelectItem>
                    {timezones.map((timezone, index) => (
                      <SelectItem key={`timezone_${index}`} value={timezone.value}>
                        {timezone.text}
                      </SelectItem>
                    ))}
                  </SelectContent>
                </Select>
              </div>
            </div>

```

`components/atoms/Select/single-select.tsx`

```typescript
import React, { useEffect, useRef, useState } from "react";
import { RiArrowDownSLine } from "react-icons/ri";

import clsx from "clsx";

import { DropdownMenu, DropdownMenuTrigger, DropdownMenuContent } from "@radix-ui/react-dropdown-menu";
import { Command, CommandGroup, CommandInput, CommandItem } from "../Cmd/command";
import Tooltip from "../Tooltip/tooltip";
interface SingleSelectProps {
  value?: string;
  onValueChange: (value: string) => void;
  placeholder?: string;
  inputPlaceholder?: string;
  options: { label: string; value: string }[];
  position?: "popper" | "item-aligned";
  isSearchable?: boolean;
  insetLabel?: string;
  labelText?: string;
}

const SingleSelect = ({
  placeholder,
  value,
  onValueChange,
  options,
  position,
  inputPlaceholder,
  isSearchable = false,
  insetLabel,
  labelText,
}: SingleSelectProps) => {
  const inputRef = useRef<HTMLInputElement>(null);
  const [inputValue, setInputValue] = useState("");
  const [isOpen, setIsOpen] = useState(false);

  const current = options.find((option) => option.value === value);

  useEffect(() => {
    if (isOpen && isSearchable) {
      // TODO: Revisit this. There's a timing issue where the input's ref is set when the dropdown is opened
      // but it still can't receive focus. This is a workaround to focus the input after
      // the dropdown is opened.
      setTimeout(() => {
        if (isOpen && isSearchable) {
          inputRef.current?.focus();
        }
      });
    }
  }, [isOpen, isSearchable]);

  return (
    <label>
      <span className="sr-only">{labelText}</span>
      <DropdownMenu
        open={isOpen}
        modal={false}
        onOpenChange={(open) => {
          setIsOpen(open);
          if (!open) {
            setInputValue("");
          }
        }}
      >
        <DropdownMenuTrigger
          data-inset-label={insetLabel}
          className={clsx(
            "flex w-full justify-between md:w-fit text-sm px-3 py-1.5 !border !border-slate-200 rounded-md bg-white data-[state=open]:border-orange-500 min-w-max",
            insetLabel && `before:content-[attr(data-inset-label)] before:mr-1 before:font-normal before:text-slate-500`
          )}
        >
          <Tooltip content={current?.label ?? placeholder}>
            <div className="flex items-center w-44 ">
              <p className="flex-grow text-start truncate">{current?.label ?? placeholder}</p>
            </div>
          </Tooltip>

          <div className="flex items-center">
            <RiArrowDownSLine size={20} className="w-5 text-slate-400" />
          </div>
        </DropdownMenuTrigger>

        <DropdownMenuContent className="!p-0 z-50 relative bg-white w-[90vw] md:w-auto my-1 border shadow-lg rounded-lg">
          <>
            {options.length > 0 && (
              <Command loop className="w-full px-0 pt-1 bg-transparent">
                {isSearchable && (
                  <CommandInput
                    ref={inputRef}
                    placeholder={inputPlaceholder ?? "Search Items"}
                    value={inputValue}
                    onValueChange={setInputValue}
                    className="px-2 focus:ring-0"
                  />
                )}
                <CommandGroup className="flex flex-col overflow-x-hidden overflow-y-scroll max-h-52">
                  {isOpen && options.length > 0
                    ? options.map((option) => (
                        <CommandItem
                          key={option.value}
                          onSelect={(value) => {
                            setInputValue("");
                            setIsOpen(false);
                            onValueChange(option.value);
                          }}
                          onMouseDown={(e) => {
                            e.preventDefault();
                            setIsOpen(false);
                            onValueChange(option.value);
                          }}
                          onClick={() => {
                            setIsOpen(false);
                            onValueChange(option.value);
                          }}
                          className="!z-50 !cursor-pointer flex justify-between min-w-[7rem] items-center !px-3 rounded-md truncate break-words w-full"
                        >
                          {option.label}
                        </CommandItem>
                      ))
                    : null}
                </CommandGroup>
              </Command>
            )}
          </>
        </DropdownMenuContent>
      </DropdownMenu>
    </label>
  );
};

export default SingleSelect;

```

`components/shared/DayRangePicker.tsx`

```typescript

return (
    <div className="w-fit font-semibold">
      <span className="sr-only">Range:</span>
      <SingleSelect
        labelText="Day range select"
        insetLabel="Range:"
        onValueChange={onDefaultDayRangeChanged}
        value={`${dayRange}`}
        options={dayRanges}
      />
    </div>
  );
  
  ```